src/Sylius/Bundle/ResourceBundle/EventListener/ResourceDeleteSubscriber.php

```diff
@@ -83,36 +84,43 @@ public static function getSubscribedEvents()
     public function onResourceDelete(GetResponseForExceptionEvent $event)
     {
         $exception = $event->getException();
-        if (!$exception instanceof ForeignKeyConstraintViolationException) {
+        if (!$exception instanceof ResourceConstraintViolationException) {
             return;
         }
 
         $requestAttributes = $event->getRequest()->attributes;
         $originalRoute = $requestAttributes->get('_route');
-        $resourceName = $this->getResourceNameFromRoute($originalRoute);
+
+        if (null === $requestAttributes->get('_controller')) {
```

Comment:
The `has` method use php `array_key_exists` method so it's only check `_controller` existence but we want to check; it's not null
[link](http://stackoverflow.com/questions/3210935/difference-between-isset-and-array-key-exists)